DataFormats/Math/interface/roundAtPrecision.h

```diff
@@ -0,0 +1,34 @@
+#ifndef DataFormatsMathroundAtPrecision_H
+#define DataFormatsMathroundAtPrecision_H
+
+#include "FWCore/Utilities/interface/isFinite.h"
+#include <cstring>
+#include <cstdint>
+
+// round with N significant bits
+template <int N>
+float roundAtPrecision(float x) {
+  if (edm::isNotFinite(x))
```

Comment:
other than this, what is the performance difference wrt reduceMantissaToNbits in libminifloat?
I'd rather not proliferate. If this is better and the isNotFinite is the only difference, perhaps it's better to update reduceMantissaToNbits

Reply:
reduceMantissaToNbits just truncate.
reduceMantissaToNbitsRounding most probably is numerically equivalent even if the code looks more complex

Reply:
but isn't this code actually a truncation?
I see the unit test ```   assert(roundAtPrecision<0>(2.9) == 2.f);``` (round(2.9) = 3)

Reply:
It is a binary rounding
with only exponent (0 bits) you have to choose between 2 and 4
so 2.9 rounds to 2 and pi to 4...